app/controllers/Application.scala

```diff
@@ -31,17 +35,54 @@ class Application @Inject()(val messagesApi: MessagesApi) extends Controller wit
       },
       answer => {
         val p = Promise[Result]()
-        p.success {
-          if (!sbtInstalled) BadRequest {
+        if (!sbtInstalled) p.success {
+          BadRequest(
             views.html.prob(task, probForm.bindFromRequest().withError("prob", "Cannot test your code now"))
-          } else {
-            Ok(Seq("pwd").!!)
+          )
+        } else Future(blocking {
+          p.success(Ok(createProjectAndTest(answer.prob)))
+        })
+        p.future
+      })
+  }
+
+  def createProjectAndTest(solution: String): String = {
+    import scala.collection.JavaConversions._
+    Try {
+      val buildSbt = List(
+        """lazy val root = (project in file("."))""",
+        """  .settings(""",
+        """    scalaVersion := "2.11.7",""",
+        """    libraryDependencies += "org.scalatest" %% "scalatest" % "2.2.6" % "test"""",
+        """  )"""
+      )
+      val d = Files.createTempDirectory("test")
+      Files.write(d.resolve("build.sbt"), buildSbt, StandardCharsets.UTF_8)
 
+      val project = d.resolve("project")
+      Files.createDirectory(project)
 
-          }
+      Files.write(project.resolve("build.properties"), List("sbt.version=0.13.9"), StandardCharsets.UTF_8)
+
+      val solutionTargetPath = d.resolve("src").resolve("main").resolve("scala")
+      Files.createDirectories(solutionTargetPath)
+      Files.write(solutionTargetPath.resolve("UserSolution.scala"), List("package tests", s"object SleepInSolution {$solution}"), StandardCharsets.UTF_8)
+
+      val testTargetPath = d.resolve("src").resolve("test").resolve("scala")
+      Files.createDirectories(testTargetPath)
+      val testSourcePath = Paths.get(app.path.getAbsolutePath, "test", "tests")
+
+      Files.walkFileTree(testSourcePath, new SimpleFileVisitor[Path] {
+        override def visitFile(file: Path, attrs: BasicFileAttributes): FileVisitResult = {
+          Files.copy(file, testTargetPath.resolve(testSourcePath.relativize(file)))
+          FileVisitResult.CONTINUE
         }
-        p.future
       })
+      val sbtCommands = Seq("sbt", "-Dsbt.log.noformat=true", "test")
+      val output = new StringBuilder
```

Comment:
Everything looks fine, just need to replace StringBuilder with something lightweight like List